src/driver/mod.rs

```diff
@@ -42,12 +42,48 @@ mod write;
 mod writev;
 
 use crate::driver::op::Lifecycle;
-use io_uring::opcode::AsyncCancel;
-use io_uring::IoUring;
+use io_uring::opcode::{self, AsyncCancel};
+use io_uring::{IoUring, Probe};
 use slab::Slab;
 use std::io;
 use std::os::unix::io::{AsRawFd, RawFd};
 
+pub(crate) fn probe() -> bool {
+    // Check we can create an io_uring instance, which effectively verifies
+    // that io_uring_setup() syscall is supported.
+    let io_uring = match IoUring::new(1) {
+        Ok(io_uring) => io_uring,
+        Err(_) => {
+            return false;
+        }
+    };
+    let submitter = io_uring.submitter();
+
+    let mut probe = Probe::new();
+
+    // Check we can register a probe to validate supported operations.
+    if let Err(_) = submitter.register_probe(&mut probe) {
+        return false;
+    }
+
+    // Check IORING_OP_FSYNC is supported
+    if !probe.is_supported(opcode::Fsync::CODE) {
```

Comment:
I just simply check if io-uring is supported, if the kernel supports io-uring, then usually  most of the opcode is supported. It might be more friendly to add an interface here to probe specific opcode.